main.go

```diff
@@ -701,16 +707,16 @@ func (e *Exporter) parseStatsSettings(ch chan<- prometheus.Metric, statsSettings
 	return parseError
 }
 
-func (e *Exporter) parseAndNewMetric(ch chan<- prometheus.Metric, desc *prometheus.Desc, valueType prometheus.ValueType, stats map[string]string, key string, labelValues ...string) error {
-	return e.extractValueAndNewMetric(ch, desc, valueType, parse, stats, key, labelValues...)
+func (e *Exporter) parseAndNewMetric(ch chan<- prometheus.Metric, desc *prometheus.Desc, valueType prometheus.ValueType, stats map[string]string, key string, logger log.Logger, labelValues ...string) error {
+	return e.extractValueAndNewMetric(ch, desc, valueType, parse, stats, key, logger, labelValues...)
 }
 
-func (e *Exporter) parseBoolAndNewMetric(ch chan<- prometheus.Metric, desc *prometheus.Desc, valueType prometheus.ValueType, stats map[string]string, key string, labelValues ...string) error {
-	return e.extractValueAndNewMetric(ch, desc, valueType, parseBool, stats, key, labelValues...)
+func (e *Exporter) parseBoolAndNewMetric(ch chan<- prometheus.Metric, desc *prometheus.Desc, valueType prometheus.ValueType, stats map[string]string, key string, logger log.Logger, labelValues ...string) error {
+	return e.extractValueAndNewMetric(ch, desc, valueType, parseBool, stats, key, logger, labelValues...)
 }
 
-func (e *Exporter) extractValueAndNewMetric(ch chan<- prometheus.Metric, desc *prometheus.Desc, valueType prometheus.ValueType, f func(map[string]string, string) (float64, error), stats map[string]string, key string, labelValues ...string) error {
-	v, err := f(stats, key)
+func (e *Exporter) extractValueAndNewMetric(ch chan<- prometheus.Metric, desc *prometheus.Desc, valueType prometheus.ValueType, f func(map[string]string, string, log.Logger) (float64, error), stats map[string]string, key string, logger log.Logger, labelValues ...string) error {
+	v, err := f(stats, key, logger)
```

Comment:
logger is available as e.logger, no need to pass it down.

Reply:
Done

Reply:
This is still around in several places.